lib/resources/shadow.rb

```diff
@@ -43,7 +42,7 @@ def initialize(path = '/etc/shadow', opts = nil)
       @filters = opts[:filters] || ''
       @raw_content = opts[:content] || read_file_content(@path, allow_empty: true)
       @lines = @raw_content.to_s.split("\n")
-      @params = @lines.map { |l| parse_shadow_line(l) }
+      params
```

Comment:
I'd really rather see this assignment preserved, rather than treating the setup as a side-effect of calling a method.

```ruby
def initialize
  @params = set_params(@lines)
end

def set_params(lines)
  lines.nil? ? [] : lines.map { |l| parse_shadow_line(1) }
end
```

Reply:
While we're there, `@raw_content` and `@lines` are only used in the initializer (if you pass `@lines` to the `params` method) so why not make them local variables instead? You could even inline the lines parsing because there's no conditional logic involved.
```
def initialize(path = '/etc/shadow', opts = nil)
  # ...
  params(@raw_content.to_s.split("\n"))
end
```

...And while we're there
```ruby
def initialize(path = '/etc/shadow', opts = nil)
  opts ||= {}
  # ...
end
```
This makes me so sad. Would you mind?
```ruby
def initialize(path = '/etc/shadow', opts = {})
  # ...
end
```

Reply:
Ah, I missed `def lines` elsewhere. Still, I'd pass it into the new method that makes your params and use it as a local variable there.